File: e2e/sut/modules/sample1/src/main/java/io/mateu/sample1/SimpleForm.java
package io.mateu.sample1;

import io.mateu.uidl.annotations.Button;
import io.mateu.uidl.annotations.Title;
import io.mateu.uidl.annotations.UI;
import io.mateu.uidl.data.Message;
import jakarta.validation.constraints.NotEmpty;

@UI("")
@Title("Simple Form")
public class SimpleForm {

    @NotEmpty
    String name;

    @Button
    public Message greet() {
        return new Message("Hello " + name + "!");
    }

}

File: backend/dotnet/samples/Mateu.Demo/SimpleForm.cs
using System.ComponentModel.DataAnnotations;
using Mateu.Uidl;

namespace Mateu.Demo;

[UI(""), Title("Simple Form")]
public class SimpleForm
{
    [Required]
    public string? Name { get; set; }

    [Button]
    public Message Greet() => new($"Hello {Name}!");
}
